src/pool/srv_target.c

```diff
@@ -99,23 +112,9 @@ ds_pool_child_put(struct ds_pool_child *child)
 {
 	D_ASSERTF(child->spc_ref > 0, "%d\n", child->spc_ref);
 	child->spc_ref--;
-	if (child->spc_ref == 0) {
-		D_DEBUG(DB_MGMT, DF_UUID": destroying\n",
-			DP_UUID(child->spc_uuid));
-		D_ASSERT(d_list_empty(&child->spc_list));
-		D_ASSERT(d_list_empty(&child->spc_cont_list));
-
-		ds_stop_chkpt_ult(child);
-		/* only stop gc ULT when all ops ULTs are done */
-		stop_gc_ult(child);
-		stop_flush_ult(child);
-
-		vos_pool_close(child->spc_hdl);
-		dss_module_fini_metrics(DAOS_TGT_TAG, child->spc_metrics);
-		ABT_eventual_set(child->spc_ref_eventual,
-				 (void *)&child->spc_ref,
+	if (child->spc_ref == 0 && *child->spc_state == POOL_CHILD_STOPPING)
```

Comment:
seems when the spc_ref drop to zero, the spc_state must be STOPPING?

Reply:
pool child cache doesn't hold pool child reference now, so the pool child reference could be dropped to zero even if it's still in cache (in start mode).